### PR TITLE
Modified downcast method for measurements passed to QuantumProgram in pyo3

### PR DESCRIPTION
**Pull request by kbarkhqs**

### CODE REVIEW

File: qoqo/src/measurements/basis_rotation_measurement.rs [hunk 1]
```diff
@@ -222,6 +222,47 @@ impl PauliZProductWrapper {
...
+        let serialized = serialize(&self.internal)
+            .map_err(|_| PyValueError::new_err("Cannot serialize PauliZProduct to bytes"))?;
+        let b: Py<PyByteArray> = Python::with_gil(|py| -> Py<PyByteArray> {
+            PyByteArray::new(py, &serialized[..]).into()
+        });
+        Ok(b)
+    }
+
+    #[allow(unused_variables)]
+    #[classmethod]
```
Comment (nfwvogt) [inline qoqo/src/measurements/basis_rotation_measurement.rs]: We should use staticmethod then we do not need cls and unused_variable

File: qoqo/src/measurements/cheated_basis_rotation_measurement.rs [hunk 2]
```diff
@@ -221,6 +221,47 @@ impl CheatedPauliZProductWrapper {
...
+        let serialized = serialize(&self.internal)
+            .map_err(|_| PyValueError::new_err("Cannot serialize CheatedPauliZProduct to bytes"))?;
+        let b: Py<PyByteArray> = Python::with_gil(|py| -> Py<PyByteArray> {
+            PyByteArray::new(py, &serialized[..]).into()
+        });
+        Ok(b)
+    }
+
+    #[allow(unused_variables)]
+    #[classmethod]
```
Comment (nfwvogt) [inline qoqo/src/measurements/cheated_basis_rotation_measurement.rs]: staticmethod

File: qoqo/src/measurements/cheated_measurement.rs [hunk 3]
```diff
@@ -221,6 +221,47 @@ impl CheatedWrapper {
...
+        let serialized = serialize(&self.internal)
+            .map_err(|_| PyValueError::new_err("Cannot serialize Cheated to bytes"))?;
+        let b: Py<PyByteArray> = Python::with_gil(|py| -> Py<PyByteArray> {
+            PyByteArray::new(py, &serialized[..]).into()
+        });
+        Ok(b)
+    }
+
+    #[allow(unused_variables)]
+    #[classmethod]
```
Comment (nfwvogt) [inline qoqo/src/measurements/cheated_measurement.rs]: staticmethod

File: qoqo/src/quantum_program.rs [hunk 4]
```diff
@@ -91,34 +91,34 @@ impl QuantumProgramWrapper {
...
-        if let Ok(try_downcast) = measurement.extract::<CheatedWrapper>() {
+        if let Ok(try_downcast) = CheatedWrapper::from_pyany(measurement.into()) {
             return Ok(Self {
                 internal: QuantumProgram::Cheated {
-                    measurement: try_downcast.internal,
+                    measurement: try_downcast,
                     input_parameter_names,
                 },
             });
         }
```
Comment (nfwvogt) [inline qoqo/src/quantum_program.rs:117]: With these changes we do not beed the following block with the _internal_to_bincode anymore.

**Review by nfwvogt — APPROVED** [review]

File: qoqo/src/quantum_program.rs (hunk 4 above)
Comment (kbarkhqs) [reply qoqo/src/quantum_program.rs:115]: I've removed it in the latest commit

File: qoqo/src/measurements/cheated_measurement.rs (hunk 3 above)
Comment (kbarkhqs) [reply qoqo/src/measurements/cheated_measurement.rs]: Changed to staticmethod

File: qoqo/src/measurements/cheated_basis_rotation_measurement.rs (hunk 2 above)
Comment (kbarkhqs) [reply qoqo/src/measurements/cheated_basis_rotation_measurement.rs]: Changed to staticmethod

File: qoqo/src/measurements/basis_rotation_measurement.rs (hunk 1 above)
Comment (kbarkhqs) [reply qoqo/src/measurements/basis_rotation_measurement.rs]: Changed to staticmethod

**Pull request merged**